src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,10 +1,97 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { validateField } from '../../services/validateField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   const [count] = useState(0);
+  const [formData, setFormData] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const [touched, setTouched] = useState({
+    title: false,
+    description: false,
+    imgUrl: false,
+    imdbUrl: false,
+    imdbId: false,
+  });
+  const [fieldValidErrors, setFieldValidErrors] = useState({
+    title: false,
+    imgUrl: false,
+    imdbUrl: false,
+    imdbId: false,
+  });
+
+  const handleTextFieldChange = (name: string, value: string) => {
+    if (!(name === 'description')) {
+      validateField(
+        name,
+        value,
+        fieldValidErrors,
+        setFieldValidErrors,
+      );
+    }
+
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const reset = () => {
+    setFormData({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+
+    setTouched({
+      title: false,
+      description: false,
+      imgUrl: false,
+      imdbUrl: false,
+      imdbId: false,
+    });
+    setFieldValidErrors({
+      title: false,
+      imgUrl: false,
+      imdbUrl: false,
+      imdbId: false,
+    });
+  };
+
+  const handleAddClick = () => {
+    const {
+      title,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = formData;
+
+    if (title && imgUrl && imdbUrl && imdbId) {
+      onAdd(formData);
+      reset();
+    }
+  };
+
+  const handleDisableButton = () => (
```

Comment:
```suggestion
  const isButtonDisabled => (
```
it might be just a const